ToDo.DataAccess/ToDo.DataAccess/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using ToDo.DataAccess.DataBase;
+using ToDo.DataAccess.Repositories.Interfaces;
+
+namespace ToDo.DataAccess
+{
+    public class GenericRepository<T> : IGenericRepository<T> where T : class,  ICategoryRepository
```

Comment:
I am expecting from you all repositories in action.

Reply:
Either remove all IXXXRepository and use generic one, or inherit all and use all of them.

Reply:
All repositories were removee and use generic one.